src/SprykerEco/Zed/PunchoutCatalogs/Business/Writer/PunchoutCatalogsWriter.php

```diff
@@ -62,14 +68,59 @@ public function updateConnection(PunchoutCatalogConnectionTransfer $punchoutCata
 
         $isSuccessful = $this->punchoutCatalogsEntityManager->updatePunchoutCatalogConnection($punchoutCatalogConnectionTransfer);
 
+        if (!$isSuccessful) {
+            return $this->createUnsuccessfulPunchoutCatalogResponseTransfer(static::MESSAGE_ERROR_DURING_CONNECTION_UPDATE);
+        }
+
+        return $this->storePassword($punchoutCatalogConnectionTransfer, static::MESSAGE_ERROR_DURING_CONNECTION_UPDATE);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\PunchoutCatalogConnectionTransfer $punchoutCatalogConnectionTransfer
+     * @param string $errorMessage
+     *
+     * @return \Generated\Shared\Transfer\PunchoutCatalogResponseTransfer
+     */
+    protected function storePassword(PunchoutCatalogConnectionTransfer $punchoutCatalogConnectionTransfer, string $errorMessage): PunchoutCatalogResponseTransfer
+    {
+        if (!$punchoutCatalogConnectionTransfer->getPassword()) {
+            return $this->createSuccessfulResponseTransfer($punchoutCatalogConnectionTransfer);
+        }
+
+        $isSuccessful = $this->vaultFacade->store(
+            static::VAULT_DATA_TYPE_PASSWORD,
+            (string)$punchoutCatalogConnectionTransfer->getIdPunchoutCatalogConnection(),
+            $punchoutCatalogConnectionTransfer->getPassword()
+        );
+
         if ($isSuccessful) {
-            return (new PunchoutCatalogResponseTransfer())
-                ->setPunchoutCatalogConnection($punchoutCatalogConnectionTransfer)
-                ->setIsSuccessful(true);
+            return $this->createSuccessfulResponseTransfer($punchoutCatalogConnectionTransfer);
         }
 
+        return $this->createUnsuccessfulPunchoutCatalogResponseTransfer($errorMessage);
+    }
+
+    /**
+     * @param string $message
+     *
+     * @return \Generated\Shared\Transfer\PunchoutCatalogResponseTransfer
+     */
+    protected function createUnsuccessfulPunchoutCatalogResponseTransfer(string $message): PunchoutCatalogResponseTransfer
```

Comment:
- `createUnsuccessfulPunchoutCatalogResponseTransfer `
- `createSuccessfulResponseTransfer `

Please keep naming consistency, and stick to 1 way or to another.

P. S. Also, both methods seem useless (no benefits of having a separate method for creating a transfer without extra logic).

Reply:
Removed `createUnsuccessfulPunchoutCatalogResponseTransfer` since it used only once, but I would like to keep `createSuccessfulResponseTransfer` just to reduce method size